### PR TITLE
Set statblock Source from description text

### LINKED ISSUE #529: Use Source from description text when importing statblock
Given a DnDAppFile or FightClube5eXML formatted statblock
When importing
Then populate the Source field from information in the element's description field.

**Background** 
In older DnDAppFile XML format, the Source was included in the type field and delimited by a comma.
E.g.
```
<type>humanoid (any race), monster manual</type>
```

In more recent DnDAppFile format, the Source is included in the description field.
E.g.
```
<type>humanoid (any race)</type>
<description>Source: Monster Manual</description>
```

And in even more recent FightClub5eXML format, the Source is included in the description field along with a description text.
E.g.
```
<description>Acolytes are junior members of a clergy, usually answerable to a priest.
Source: Monster Manual p. 342, Curse of Strahd, Hoard of the Dragon Queen, Princes of the Apocalypse
</description>
```

### PR DESCRIPTION
**Pull request by mphartman**

### CODE REVIEW

File: client/Importers/StatBlockImporter.ts [hunk 1]
```diff
@@ -4,19 +4,53 @@ import { AccountClient } from "../Account/AccountClient";
...
+    const description = this.getString("description");
+    const searchString = "Source: ";
+    const sourcePos = description.lastIndexOf(searchString);
+    if (sourcePos != -1) {
+      const sources = description
+        .substr(sourcePos + searchString.length)
+        .split(/, ?/);
+      source = sources[0];
+    } else {
+      const typeString = this.getString("type");
```
Comment (cynicaloptimist) [inline client/Importers/StatBlockImporter.ts:47]: Could you still use `getCommaSeparatedStrings` in this else block?

**Review by cynicaloptimist — CHANGES_REQUESTED** [review]
Looks pretty good! Will merge after feedback is addressed. Thanks!

File: client/Importers/StatBlockImporter.ts (hunk 1 above)
Comment (mphartman) [reply client/Importers/StatBlockImporter.ts:45]: It's possible. So something like this?
```
      const types = this.getCommaSeparatedStrings("type");
      source = _.startCase(types.length > 1 ? types[types.length - 1] : "");
```
Need to check that there's at least 2 elements in the array in order to satisfy the test `Type should ignore missing source from type field`

Comment (cynicaloptimist) [reply client/Importers/StatBlockImporter.ts:45]: Yeah, good catch. Your suggestion works great.

**Comment by cynicaloptimist** [comment]
Nice work, thanks Michael!

**Pull request merged**